File: Code/Back-end/src/main/java/com/example/harsha/service/MessagesService.java
package com.example.harsha.service;

import java.util.List;

import com.example.harsha.entity.Messages;

public interface MessagesService {
	public List<Messages> getMessages(String sendTo);
	public List<Messages> getSentMessages(String sendFrom);
	public List<Messages> getMessagesByBatch(String batch);
	public String addMessage(Messages messages);
	
}

File: Code/Front-end/src/app/messages.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import { from } from 'rxjs';
@Injectable({
  providedIn: 'root'
})
export class MessagesService {
  baseUrl="http://localhost:8080/";
  constructor(private httpClient:HttpClient) { }

  public addMessage(message){
    return this.httpClient.post(this.baseUrl+"addMessage",message,{responseType:"text" as "json"});
  }
  public getMessages(email){
    return this.httpClient.get(this.baseUrl+"getMessages/"+email);
  }
  public getSentMessages(email){
    return this.httpClient.get(this.baseUrl+"getSentMessages/"+email);
  }
  public getessagesByBatch(batch){
    return this.httpClient.get(this.baseUrl+"getMessagesByBatch/"+batch);
  }
}